### PR TITLE
Fix errors when running the test on Apple Silicon M1

### PR DESCRIPTION
**Pull request by tenzen-y**
**What this PR does / why we need it**:
I fixed the following errors when running the test on Apple Silicon M1.

> unable to find a version that was supported for platform darwin/arm64

It seems that this error occurred when `setup-envtest` tried to download the Kubernetes components binary for darwin/arm64 as described in [this](https://github.com/kubernetes-sigs/controller-runtime/issues/1657) issue. 
So I specified CPU architecture when running the `setup-envtest`.

> \# github.com/shirou/gopsutil/process
../../../../pkg/mod/github.com/shirou/gopsutil@v2.20.7+incompatible/process/process_darwin.go:590:34: undefined: KinfoProc
../../../../pkg/mod/github.com/shirou/gopsutil@v2.20.7+incompatible/process/process_darwin.go:591:8: undefined: KinfoProc
../../../../pkg/mod/github.com/shirou/gopsutil@v2.20.7+incompatible/process/process_darwin.go:604:32: undefined: KinfoProc
../../../../pkg/mod/github.com/shirou/gopsutil@v2.20.7+incompatible/process/process_darwin.go:608:62: undefined: KinfoProc

It seems that this error occurred since darwin/arm64 is not supported by the `github.com/shirou/gopsutil` v2 as described in [this](https://github.com/shirou/gopsutil/issues/1000) issue.
So I upgraded the `github.com/shirou/gopsutil` version to v3.

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes #

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

**Comment by tenzen-y** [comment]
We need to merge this PR after merging https://github.com/kubeflow/katib/pull/1882.
/hold

**Review by gaocegege — APPROVED** [review]
LGTM :+1:
/lgtm

**Comment by tenzen-y** [comment]
I have fixed an error from shellcheck.
https://github.com/kubeflow/katib/runs/6750242050?check_suite_focus=true

File: test/e2e/v1beta1/scripts/gh-actions/build-load.sh
Comment (johnugeorge) [inline test/e2e/v1beta1/scripts/gh-actions/build-load.sh:64]: How was this error not caught in the other PR?

Comment (tenzen-y) [reply test/e2e/v1beta1/scripts/gh-actions/build-load.sh:64]: We have fixed shellcheck command in https://github.com/kubeflow/katib/pull/1885 and https://github.com/kubeflow/katib/pull/1884.
But I forgot to rebase https://github.com/kubeflow/katib/pull/1882 after merging their PRs.

So, we could not catch this error in https://github.com/kubeflow/katib/pull/1882.

**Comment by tenzen-y** [comment]
@johnugeorge @gaocegege Thanks for your review!
/hold cancel

**Pull request merged**
